sdgym/datasets.py

```diff
@@ -83,19 +94,36 @@ def load_tables(metadata):
     return real_data
 
 
-def get_available_datasets(bucket=None):
+def get_available_datasets(bucket=None, aws_key=None, aws_secret=None):
     bucket_url = BUCKET_URL.format(bucket or BUCKET)
-    response = urllib.request.urlopen(bucket_url)
-    tree = ElementTree.fromstring(response.read())
-    datasets = []
-    for content in tree.findall('{*}Contents'):
-        key = content.find('{*}Key').text
-        size = int(content.find('{*}Size').text)
-        if key.endswith('.zip'):
-            datasets.append({
-                'name': key[:-len('.zip')],
-                'size': size
-            })
+    if S3_PREFIX in bucket_url:
```

Comment:
Same as above, the old `urllib` based approach can be removed, and `boto3` always used for the download and listing of datasets.